samza-core/src/main/java/org/apache/samza/coordinator/stream/CoordinatorStreamSystemProducer.java

```diff
@@ -82,6 +82,7 @@ public void start() {
     }
     log.info("Starting coordinator stream producer.");
     systemProducer.start();
+    systemAdmin.start();
```

Comment:
Should be started and stopped by the owner of this SystemAdmin, not this object.

Reply:
This is started here for the same reason as `CoordinatorStreamSystemConsumer`. I think it may be reasonable to remove `CoordinatorStreamSystemFactory` and just instantiate SystemAdmin in CoordinatorStreamSystemProducer directly. Does this sound OK?

Reply:
CoordinatorStreamSystemProducer constructor now takes config as the parameter.